### PR TITLE
[Connect] Reorganize approach to cluster names

### PR DESCRIPTION
**Pull request by gzdunek**
Closes https://github.com/gravitational/webapps.e/issues/261

I decided to not add `profile_name` to the API as we can extract it from the cluster URI.
Also, this field makes sense only for the root clusters and IMO we should try to minimize the differences in API between leaf and root clusters.  

>We want to steer away from code that makes the assumption that rootClusterId is included in the URI.
>In general, although I'm not 100% sure about it, I think we should avoid extracting any piece of info from URIs altogether. I have a hunch that treating them as pure identifiers only will make code around them easier to maintain, though at the moment I can't quite explain why.
>OTOH on the web you regularly extract info from URLs, so idk.

Hmm that last argument makes sense for me.
I think I'd stay with the current way of using URIs for now, until we are 100% sure that we want to change it?

### CODE REVIEW

**Review by ravicious — COMMENTED** [review]
The changes look fine and make sense overall, I just didn't have time to test this today and look closer at the frontend changes.

File: lib/teleterm/clusters/storage.go [hunk 1]
```diff
@@ -221,10 +226,8 @@ func (s *Storage) fromProfile(profileName, leafClusterName string) (*Cluster, er
 	}
 
 	return &Cluster{
-		URI: clusterURI,
-		// TODO(ravicious): This should probably use leafClusterName if available, but at this point I'm
-		// worried that changing it might break something else in the app.
-		Name:          profileName,
+		URI:           clusterURI,
+		Name:          clusterClient.SiteName,
```
[the remark this answers is not in the document]
Comment (alex-kovoy) [reply lib/teleterm/clusters/storage.go:231]: The [URIs](https://en.wikipedia.org/wiki/Uniform_Resource_Identifier) was introduced on purpose to make it easier to work with Teleport resources. These are self-descriptive IDs and provide unique path to the resource within specific Teleport cluster domain (there is [always a rootCluster](https://goteleport.com/how-it-works/) which is your edge proxy). 

Technically it is possible to access a resource such as:
(your edge proxy) Root Cluster  -> Trusted Cluster -> Trusted Cluster -> Node

This is similar to Web URLs (see Teleport Web) but for `teleterm` purposes.

I hope it helps.

[the remark this answers is not in the document]
Comment (gzdunek) [reply lib/teleterm/clusters/storage.go:231]: @ravicious I'm pretty sure it is set here https://github.com/gravitational/teleport/blob/61c39a9e2b9dda96d4032598792fe291e2a1f91b/lib/client/api.go#L3283

I'll put a comment with a link to it.

[the remark this answers is not in the document]
Comment (gzdunek) [reply lib/teleterm/clusters/storage.go:231]: >The [URIs](https://en.wikipedia.org/wiki/Uniform_Resource_Identifier) was introduced on purpose to make it easier to work with Teleport resources. These are self-descriptive IDs and provide unique path to the resource within specific Teleport cluster domain (there is [always a rootCluster](https://goteleport.com/how-it-works/) which is your edge proxy).

Yeah, I'd leave the URIs as they are.

**Review by ravicious — APPROVED** [review]
Left comments about minor stuff but looks good overall.

File: lib/teleterm/api/proto/v1/cluster.proto [hunk 2]
```diff
@@ -23,13 +23,6 @@ message Cluster {
     // uri is the cluster resource URI
     string uri = 1;
     // name is used throughout the Teleport Connect codebase as the cluster name.
-    // Unfortunately, we made a mistake where instead of taking the name of the cluster from the
-    // key, we take it from the hostname part of the proxy service. These two are not always equal.
-    // And actually there are places in tsh code where it needs the name of the cluster from the
-    // key, not the hostname of the proxy service (like in the `TELEPORT_CLUSTER` env var). Since
-    // this identifier is so widely used, we didn't have time to change it before the preview
-    // release. In the future we plan to fix the value sent under name to the name from the key and
-    // get rid of the `actual_name` field.
```
[the remark this answers is not in the document]
Comment (gzdunek) [reply lib/teleterm/api/proto/v1/cluster.proto:32]: Done, we have comments only in the Go version.

File: lib/teleterm/clusters/dbcmd_cli_command_provider.go [hunk 3]
```diff
@@ -63,7 +63,7 @@ func (d DbcmdCLICommandProvider) GetCommand(gateway *gateway.Gateway) (string, e
 		// At this point it doesn't matter though because this argument is used only for
 		// generating correct CA paths. We use dbcmd.WithNoTLS here which means that the CA paths aren't
 		// included in the returned CLI command.
-		cluster.GetActualName(),
+		cluster.Name,
```
[the remark this answers is not in the document]
Comment (gzdunek) [reply lib/teleterm/clusters/dbcmd_cli_command_provider.go:66]: Done

**Review by kimlisa — APPROVED** [review]
bot

**Review by rudream — APPROVED** [review]

**Pull request merged**